src/main/java/org/dave/compactmachines3/block/BlockWall.java

```diff
@@ -104,8 +104,9 @@ public boolean onBlockActivated(World world, BlockPos pos, IBlockState state, En
                 playerStack.setCount(playerStack.getCount()-1);
                 WorldSavedDataMachines.getInstance().addTunnel(pos, tunnelSide);
             } else {
-                // TODO: Localization
-                player.sendStatusMessage(new TextComponentString(TextFormatting.RED + "All tunnels already used up"), false);
+                TextComponentTranslation tc = new TextComponentTranslation("hint.compactmachines3.all_tunnels_used");
+                tc.getStyle().setColor(TextFormatting.RED);
+                player.sendStatusMessage(tc, false);
```

Comment:
If I recall, the boolean here changes the message to display above the hotbar - we should prefer that so we aren't spamming the chat list with messages.

Reply:
Makes sense, will do.